### PR TITLE
update action

### PR DESCRIPTION
**Pull request by melindaloubser1**
Bump the Dockerfile & Github Actions definition to align with the changes introduced for Rasa 3.x compatibility here: https://github.com/RasaHQ/nlu-hyperopt/pull/18

### CODE REVIEW

**Review by ancalita — COMMENTED** [review]
@melindaloubser1 Could you please add a brief description of the changes you made and why were they necessary, slightly clueless about what I'm looking at 🙈

File: .github/workflows/nlu-hyperopt.yml [hunk 1]
```diff
@@ -8,12 +8,19 @@ jobs:
...
+    - name: Cancel Previous Runs
+      uses: styfle/cancel-workflow-action@0.8.0
+      with:
+        access_token: ${{ github.token }}
+
+    - name: Checkout pull request HEAD commit instead of merge commit
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+    - uses: ./
```
Comment (ancalita) [inline .github/workflows/nlu-hyperopt.yml:20]: Potentially the `README.md` needs to be updated to align with these workflow changes?

**Comment by melindaloubser1** [comment]
done, apologies should have done that first

File: .github/workflows/nlu-hyperopt.yml (hunk 1 above)
Comment (melindaloubser1) [reply .github/workflows/nlu-hyperopt.yml:20]: You're right, I bumped the version (release will happen on merge of this PR) and updated the default data dir there.

**Review by ancalita — APPROVED** [review]
🎸

**Pull request merged**